src/app/space/[did]/root/[cid]/shard/[shard]/page.tsx

```diff
@@ -146,3 +178,96 @@ function isPieceLink(link: any): link is PieceLink {
     return false
   }
 }
+
+const MAX_DEPTH = 63
+
+// Adapted from https://github.com/web3-storage/data-segment/blob/e9cdcbf76232e5b92ae1d13f6cf973ec9ab657ef/src/proof.js#L62-L86
+function InclusionProof ({ proof, piece, style }: { proof: ProofData, piece: PieceView, style: ProofStyle }): JSX.Element {
```

Comment:
it'd be nice to extract some of the logic in here to hooks and/or sub-components, tho I would definitely buy that the various variables are all complected together in a way that makes that hard, in which case: carry on!